docs: Updates commands in quickstart

vikram-dagger opened the pull request:
This commit updates the current Zenith quickstart to use the new `dagger call` syntax. It is superseded by #6262 .

vikram-dagger commented:
@sipsma I ran through these commands and came across 4 function calls that fail with the example app. In all cases, the error is the same

```
┃ npm ERR! enoent ENOENT: no such file or directory, open '/src/package.json'                  
┃ npm ERR! enoent This is related to npm not being able to find a file.      
```

Could you please have a look? It may be that the example module code in https://github.com/dagger/hello-dagger needs an update. The failing commands are marked inline.

vikram-dagger commented on docs/versioned_docs/version-zenith/user/quickstart/292861-export.mdx:
````diff
@@ -0,0 +1,40 @@
+To see it in action, use `export` with the `build` function used previously, to export the built application to an `out` directory on the host:
+
+```shell
+dagger call build export --path ./out
````
Fails

vikram-dagger replied:
Fixed now

vikram-dagger commented on docs/versioned_docs/version-zenith/user/quickstart/292861-export.mdx:
````diff
@@ -0,0 +1,40 @@
+The `export` API also works with functions that return `Container` types, in which case the `Container` is exported as a TAR file to the host file system. Here's an example:
+
+```shell
+dagger call app-container export --path ./container.tar
````
Fails

vikram-dagger replied:
Yes, I did so already https://github.com/dagger/dagger/pull/6548#issuecomment-1921052528 not

vikram-dagger replied:
Fixed now

vikram-dagger commented on docs/versioned_docs/version-zenith/user/quickstart/302010-shell.mdx:
````diff
@@ -32,12 +30,12 @@ Linux vtmeopf0atpc2.3jgto8svt9r6s.dagger.local 6.2.0-34-generic #34-Ubuntu SMP P
 
 ```shell
-dagger shell --entrypoint node debug
+dagger call debug shell --args node
````
Fails

vikram-dagger replied:
Fixed now

vikram-dagger commented on docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:
````diff
@@ -21,7 +21,7 @@ In order for this to work, the service returned by the function must have the `C
 
 ```shell
-dagger up --native service
+dagger call service up --native
````
Fails

vikram-dagger replied:
Fixed now

vikram-dagger commented on docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:
````diff
@@ -33,14 +33,14 @@ The `--native` flag results in the exposed ports on the container being auto-map
 
 ```shell
-dagger up --port 9090:8080 service
+dagger call service up --ports 9090:8080
````
Fails

vikram-dagger replied:
Fixed now

vikram-dagger commented on docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:
```diff
@@ -33,14 +33,14 @@ The `--native` flag results in the exposed ports on the container being auto-map
 Here is an example of using this command with a module from the [Daggerverse](https://daggerverse.dev/):
 
-- `dagger up --native --mod github.com/helderco/daggerverse/python@kcna23 http-server --src <path/to/web/assets>`
+- `dagger call -m github.com/helderco/daggerverse/python@kcna23 http-server up --src <path/to/web/assets> --native`
```
Fails, for @helderco 
```
┃ ImportError: cannot import name 'Annotated' from 'dagger.mod' 
```

helderco replied:
Remove `@kcna23`. That's fixed to last kubecon.

helderco replied:
Also, `up` is in the wrong place. `--src` is for `http-server` function:
```suggestion
- `dagger call -m github.com/helderco/daggerverse/python http-server --src <path/to/web/assets>  up --native`
```

vikram-dagger replied:
Fixed

sipsma replied:
@vikram-dagger did you forget to merge the fixes here? The diff I'm seeing for the PR still shows `dagger call -m github.com/helderco/daggerverse/python@kcna23 http-server up --src <path/to/web/assets> --native`

helderco replied:
I see it.

gerhard commented:
I **think** that we want to merge #6262 first, then this one. OK to do the same here too: https://github.com/dagger/dagger/pull/6262#issuecomment-1921463759

vikram-dagger commented:
> I **think** that we want to merge #6262 first, then this one. OK to do the same here too: [#6262 (comment)](https://github.com/dagger/dagger/pull/6262#issuecomment-1921463759)

If #6262 is merged, then this PR becomes irrelevant. In case it's not, then we should merge this one because it does a surface update to the current quickstart for the new `dagger call` syntax, to ensure that docs are not broken on release.

helderco commented:
@vikram-dagger, I think the failures you're getting mean the hello-dagger repo needs fixing. Possibly this dependency: https://github.com/dagger/hello-dagger/blob/438911579b2e8ee2e1d2172ee347630120553d4a/dagger.json#L5

```
json: error calling MarshalJSON for type *main.Directory: input: resolve: node: withVersion: withNpm: withSource: install: build: container: directory: process "npm install" did not complete successfully: exit code: 254
input: resolve: node: withVersion: withNpm: withSource: install: build: container: directory: process "npm install" did not complete successfully: exit code: 254
```

vikram-dagger commented:
> @vikram-dagger, I think the failures you're getting mean the hello-dagger repo needs fixing. Possibly this dependency: https://github.com/dagger/hello-dagger/blob/438911579b2e8ee2e1d2172ee347630120553d4a/dagger.json#L5

It wasn't the dependency but I found and fixed the issue: https://github.com/dagger/hello-dagger/commit/59baf44aa28e78481ca4ecc3a01b4d69ff255dd3
 
> There's also the changes to `dagger.json`.

I've added these as well, thanks!
